### ISSUE TITLE
How to allow cell edit only after pressing "Enter", "F2", or double clicking?

### ISSUE THREAD

**Issue by ahlunju** [handsontable/handsontable#3950]
Hi,

I want to prevent user from accidentally modifying a cell when he navigate around the grid, I want user clicks Enter or F2 or double clicking before editing the cell value.

Thanks

**Comment by AMBudnik** [comment]
Hi @ahlunju 
I guess that the easiest way would be to set your data as `readOnly: true` and add a listener on keys to switch between read only modes.

Here's an example: http://jsfiddle.net/gL30doew/1/

1. I am setting `readOnly: true` to a whole grid
2. Then setting the `beforeKeyDown` to set `readOnly: false` after it gets `keyCode === 13` (<kbd>ENTER</kbd> key). Here I am using `getSelected()` methid to get `row` and `col` coordinates.
3. Use `afterChange` hook with `setCellMeta()` method to set `readOnly: true`. If we won't do it the cell will stay editable after changing data.

ps. you need to remember to call `render()` method to refresh the read only state.

**Comment by AMBudnik** [comment]
If you would have any other questions please do not hesitate to leave a comment.
